Headlines/src/Controllers/TrendingCardsViewController.swift

```diff
@@ -138,9 +175,39 @@ class TrendingCardsViewController: UICollectionViewController {
         
         //  Had to set content offset because of UIRefreshControl bug
         //  http://stackoverflow.com/a/31224299/994129
-        collectionView?.contentOffset = CGPoint(x: 0, y:-refreshCtrl.frame.size.height)
+        collectionView?.contentOffset = CGPoint(x: 0, y: -refreshCtrl.frame.size.height)
+        
+        if let appDelegate = UIApplication.shared.delegate as? AppDelegate {
+            self.newsDataTask = appDelegate.newsDataTask
+            
+            if let n = appDelegate.newsFetched {
+                self.topics.append(contentsOf: n)
+            }
+        }
+        
+        if topics.count == 0 {
+            startRefreshing()
+            
+            if self.newsDataTask?.state != .running {
+                fetchTrendingTopics()
+            }
+        }
         
-        fetchTrendingTopics()
+        let notification = Notification.Name(rawValue: "trendingTopicFinished")
+        let nc = NotificationCenter.default
+        nc.addObserver(forName: notification, object: nil, queue: nil, using: catchNotification)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/e18ab20bbfd6a83cc1d7cad460719fa49e2cf5ff/Headlines/src/Controllers/TrendingCardsViewController.swift#L198"></span>When registering for a notification using a block, the opaque observer that is returned should be stored so it can be removed later.</td>
    </tr>
  </tbody>
</table>